### PR TITLE
move it from DeviceGroup end point to imporve performance

### PR DESCRIPTION
**Pull request by mgold1234**
pacakgeDiff appear every time we get deviceGroup and is affect on th…e performance,

so I move it from DeviceGroup end point to imporve performance related to issue THEEDGE-2914

Signed-off-by: mgold1234 <mgold@redhat.com>

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

What is it?

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [ ] Refactor

# Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I run `make pre-commit` to check fmt/vet/lint/test-no-fdo

### CODE REVIEW

File: pkg/services/devicegroups.go [hunk 1]
```diff
@@ -183,7 +183,6 @@ func (s *DeviceGroupsService) GetDeviceImageInfo(images map[int]models.DeviceIma
 			var updAvailable bool
 			var deviceImage models.Image
 			var deviceImageSet models.ImageSet
-			var imagePackageDiff models.PackageDiff
```
Comment (de1987) [inline pkg/services/devicegroups.go:186]: did you check if it will affect the frontend?

Comment (ldjebran) [reply pkg/services/devicegroups.go:186]: I agree, this a service function that may be used in many different scenarios, may be we should create a device-groups-view  like we did for devices and for image-sets , that view will be available and serve only the UI needs.

Comment (ldjebran) [reply pkg/services/devicegroups.go:186]: but if that functions was created for only this UI purpose , we can go with this fix.

Comment (mgold1234) [reply pkg/services/devicegroups.go:186]: I checked and this function GetDeviceImageInfo used only once on getDeviceGroups function

**Review by de1987 — APPROVED** [review]

**Comment by omaciel** [comment]
So no tests were affected... because we didn't have any???

**Comment by mgold1234** [comment]
@omaciel we didn't check this specific field in the unit test,
and I checked the frontend side with Demetris and we see there is no affect on UI side

**Comment by omaciel** [comment]
@mgold1234 great opportunity to add a test then, don't you think? Think about it: you have just deleted a few lines of code and we have no way to tell if this will break our code...

**Comment by mgold1234** [comment]
@omaciel good point, will add it

**Review by omaciel — APPROVED** [review]
/lgtm

**Review by ldjebran — APPROVED** [review]

**Pull request merged**